docs/ruleset_functions/normalize_interior_lighting_schedules.md

```diff
@@ -30,7 +31,7 @@ Logic:
 
     - Add lighting power per area to space total: `space_total_power_per_area += power_per_area`
 
-  - Check if lighting uses schedule to model occupancy control, get occupancy control credit: `if interior_lighting.are_schedules_used_for_modeling_occupancy_control: control_credit = data_lookup(table_G3_7, space.lighting_space_type, interior_lighting.occupancy_control_type)`
+  - Check if adjust_for_credit is true and lighting uses schedule to model occupancy control, get occupancy control credit: `if adjust_for_credit and interior_lighting.are_schedules_used_for_modeling_occupancy_control: control_credit = data_lookup(table_G3_7, space.lighting_space_type, interior_lighting.occupancy_control_type)`
```

Comment:
I don't see this function checking interiorlighting.occupancy_control_type, but I beleive you showed me in the code that this is verified. Also, I don't beleive we resolved the question whether bonus credit (1.25 adjustment) is allowed for any space type or just for certain space types. I think it should be any space type.

Reply:
We did not have a data_lookup function RDS. Instead, each table search has its own implementation.
In `table_g3_7_lookup` function, the [code](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/data_fns/table_G3_7_fns.py#L167-L177) checks if the `occupancy_control_type` is one of [enum](LightingOccupancyControlOptions). Also, there is an exception handling function that will set the output to UNDETERMINED if the `occupancy_control_type` is not available when `are_schedules_used_for_modeling_occupancy_control` is `True`

As bonus credit goes, I thought the discussion was that the code and database takes care of the bonus credit.
Line 167 - 171 in `table_g3_7_lookup`
```python
 manon_or_partauto = osstd_entry["manon_or_partauto"]
    if occupancy_control_type in [PARTIAL_AUTO_ON, MANUAL_ON]:
        if manon_or_partauto:
            control_credit = osstd_entry["occup_sensor_savings"]
        else:
            control_credit = osstd_entry["occup_sensor_auto_on_svgs"]
    elif occupancy_control_type == FULL_AUTO_ON:
        control_credit = osstd_entry["occup_sensor_auto_on_svgs"]
    else:
        # no credit
        control_credit = 0.0
```

The logic is if the `interior_lighting.occupancy_control_type` is one of the `[PARTIAL_AUTO_ON, MANUAL_ON]`, then we will get the value from `occup_sensor_savings` as control credit.
The `occup_sensor_savings` is the credit that included the bonus adjustment in the [database](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/data/ashrae_90_1_prm_2019.prm_interior_lighting.json#L1238-L1250):
```json
 {
      "template": "90.1-PRM-2019",
      "lpd_space_type": "office - enclosed <= 250 sf",
      "primary_space_type": "Office",
      "secondary_space_type": "Enclosed",
      "w/ft^2": 1.1,
      "w/ft": null,
      "isresidential": 0,
      "manon_or_partauto": 1,
      "occup_sensor_auto_on_svgs": 0.3,
      "occup_sensor_savings": 0.375,
      "notes": null
    },
```
In short, the `data_lookup` described in the RDS has the capabilities of automatically apply bonus credit based on the `interior_lighting.occupancy_control_type`.

Let me know if you think the implementation shall be different than the current logic.

Reply:
I understand that the logic is implemented via expanded table G3.7, and this is totally fine just makes it a little harder to review the RDS. Few comments:
1. I think most BEM tools will not populate "occupancy_control_type", so producing UNDETERMINED outcome if the occupancy_control_type is missing when are_schedules_used_for_modeling_occupancy_control is True will result in flagging majority of projects. Since these controls are mandatory, I suggest that in this case we assume that the required controls are specified (OS reduction in 90.1 Table G3.7 should be applied) but not give the bonus credit (by multiplying adjustment by 1.25). @supriyagoel please chime in if you agree. 

2. I think we agreed that the bonus credit would be allowed for all space types, but it doesn't seem to be the case - e.g. for Attic (copied below)  "occup_sensor_auto_on_svgs"="occup_sensor_savings"=0.1. I think credit should be allowed for any space type that has the controls. Also, "Attic" space type is not in G3.7, so where 0.1 adjustment is coming from?

     "template": "90.1-PRM-2019",
      "lpd_space_type": "attic - unoccupied",
      "primary_space_type": "Attic",
      "secondary_space_type": "General",
      "w/ft^2": 0.0,
      "w/ft": null,
      "isresidential": 0,
      "manon_or_partauto": 0,
      "occup_sensor_auto_on_svgs": 0.1,
      "occup_sensor_savings": 0.1,
      "notes": null

Reply:
1. I can adjust the code to take assume NONE `occupancy_control_type` in the case where the type is missing. Will that work?
2. I was consulting the OSSTD team and seems my understanding was incorrect in some portions. I am not 100% understanding how the data is structured but below is the feedback I got from the team: 

> `manon_or_partauto = 0` means that neither Manual ON nor Partial Automatic ON are required in Chapter 9 of 90.1-2019. I think the idea was that if those are required (=1), then they need to be in the Appendix G Proposed model. If those are in the proposed model (propose is adjusted by occupancy sensor value), then the baseline is calculated from the proposed schedule based the Appendix G lighting table occupancy sensor values with adjustment of 1.25. If they are not in the proposed model, then Appendix G table occupancy sensor values applied. The column `occup sensor savings` accounts for the `manon_or_partauto`. You can see where that value is one, the `occup sensor savings` col is greater than the `occup sensor auto on svgs` col by factor of 1.25.

Based on the description, I think the logic can be simplified to:

```python
    if occupancy_control_type in [PARTIAL_AUTO_ON, MANUAL_ON, FULL_AUTO_ON]:
            control_credit = osstd_entry["occup_sensor_savings"]
    else:
        # no credit
        control_credit = 0.0
```

Is that OK?

The Attic seems using the credit from Atrium. It seems common for BEM software to do that because they have to work with much broader range of space types than table G3.7. However, inside the RCT, it will never search this data record because the `space.lighting_space_type` does not have `Attic` space type.

Reply:
Weili, I suggest assuming occupancy_control_type=Other when control is not specified. This will result in using base adjustment in Table G3.7 but not the bonus adjustment. 

I think the logic should be as follows:
if occupancy_control_type in [PARTIAL_AUTO_ON, MANUAL_ON, FULL_AUTO_ON]:
            control_credit = osstd_entry["occup_sensor_savings"]
    else:
        ****default credit from Table G3.7****
        control_credit = osstd_entry[**"occup_sensor_auto_on_svgs"**]